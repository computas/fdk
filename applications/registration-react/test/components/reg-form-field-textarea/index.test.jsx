import React from 'react';
import { shallow } from 'enzyme';
import TextAreaField from '../../../src/components/reg-form-field-textarea';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {
    showLabel: true,
    input: {
      name: 'textarea',
      value: 'Datakatalog for RAMSUND OG ROGNAN REVISJON'
    },
    label: 'label',
    type: 'textarea',
    meta: {
      active: false,
      asyncValidating: false,
      autofilled: false,
      dirty: false,
      form: 'catalog',
      initial: 'Datakatalog for RAMSUND OG ROGNAN REVISJON',
      invalid: false,
      pristine: true,
      submitting: false,
      submitFailed: false,
      touched: false,
      valid: true,
      visited: false
    }
  };
  wrapper = shallow(<TextAreaField {...defaultProps} />);
});

test('should render TextAreaField correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
