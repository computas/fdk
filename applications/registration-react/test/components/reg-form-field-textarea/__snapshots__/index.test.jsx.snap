// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TextAreaField correctly 1`] = `
<div
  className="pl-2"
>
  <label
    className="fdk-form-label w-100"
    htmlFor="textarea"
  >
    label
    <textarea
      className="form-control"
      name="textarea"
      rows="5"
      type="textarea"
      value="Datakatalog for RAMSUND OG ROGNAN REVISJON"
    />
  </label>
</div>
`;
